code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -197,6 +203,21 @@ public void handleAnalyticsTrackEvent(final Event event) {
     private void optOut() {
         Log.debug(LOG_TAG, "Privacy status is opted out, clearing event queue.");
         eventQueue.clear();
+
+        //remove AID and VID from data store.
+        if(platformServices == null){
+            Log.debug(LOG_TAG, "optout - can't remove data store values. Platformservice is null.");
+            return;
+        }
+        final LocalStorageService.DataStore dataStore = platformServices.getLocalStorageService().getDataStore(AnalyticsConstants.DataStoreKeys.ANALYTICS_DATA_STORAGE);
+        if (dataStore != null) {
+            dataStore.remove(AnalyticsConstants.DataStoreKeys.AID_KEY);
+            dataStore.remove(AnalyticsConstants.DataStoreKeys.VID_KEY);
+            Log.debug(LOG_TAG, "optout - Removed AID and VID from datastore on optout.");
+            return;
+        }
+
+        Log.debug(LOG_TAG, "optout - Failed in removing AID and VID from datastore on optout. DataStore is null.");
```

Comment:
Let us add it inside else block or switch the if statement to check `dataStore == null` and return.